core/src/main/java/com/linecorp/armeria/common/stream/StreamMessages.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.stream;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.concurrent.EventExecutor;
+
+/**
+ * A utility class for {@link StreamMessage}.
+ */
+@UnstableApi
+public final class StreamMessages {
+
+    /**
+     * Write the specified {@link  StreamMessage} to the given {@link Path} with {@link OpenOption}.
+     * If the {@link OpenOption} is not specified, defaults to {@link StandardOpenOption#CREATE},
+     * {@link StandardOpenOption#TRUNCATE_EXISTING} and {@link StandardOpenOption#WRITE}.
+     *
+     * @param publisher the source of {@link HttpData} to be written
+     * @param destination the {@link Path} to write to
+     * @param options the {@link OpenOption} specifying how the file is opened
+     * @return a {@link CompletableFuture} that completes with the specified {@link Path} or an error
+     */
+    public static CompletableFuture<Path> writeTo(StreamMessage<? extends HttpData> publisher,
```

Comment:
I realized that @kojilin forked this PR and use it in his multipart PR.
How about changing the return type to `CompletableFuture<Void>`?

Reply:
Either is good to me.